### PR TITLE
Install python virtual environment with contributed ruamel.yaml

### PR DESCRIPTION
**Pull request by awnawab**
This PR contributes the installation of a python virtual environment that contains the contributed fypp and ruamel.yaml modules. A minimal `fckit_yaml_reader` module is also installed that creates an API consistent with the pyyaml `yaml.safe_load` and `yaml.dump` functions. The aim is to provide yaml parsing functionality for offline builds of the ifs-bundle and ecwam-bundle. Downstream projects will therefore rely on either pyyaml+fypp or fckit.

I have tested the `fckit_yaml_reader` with ecwam, as a bundle build and as an externally built or installed project. I have also run the downstream_fypp tests by compiling with eckit.

### CODE REVIEW

**Comment by wdeconinck** [comment]
It looks like we need to support CMake 3.18

File: cmake/fckit_install_venv.cmake [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+# granted to it by virtue of its status as an intergovernmental organisation nor
+# does it submit to any jurisdiction.
+
+macro( fckit_install_venv )
+
+    # Discover only system install Python 3
+    set( Python3_FIND_VIRTUALENV STANDARD )
+    find_package( Python3 COMPONENTS Interpreter REQUIRED )
+    
+    # Create a loki virtualenv
```
Comment (wdeconinck) [inline cmake/fckit_install_venv.cmake:15]:
```suggestion
    # Create a virtualenv
```

**Comment by awnawab** [comment]
I've downgraded the minimum cmake to 3.17 and the minimum python to 3.6. This should fix the tests for everything except for the ubuntu gnu@ubuntu-22.04 tests. This runner seems to have a very barebones python installation:

```
  The virtual environment was not created successfully because ensurepip is not
  available.  On Debian/Ubuntu systems, you need to install the python3-venv
  package using the following command.
  
      apt install python3.10-venv
```

I think to fix this error we will have to update to the runner config.

**Comment by awnawab** [comment]
As discussed offline, I've made the build and installation of the python virtual environment optional. This defaults to off, and if left disabled then we just install the fypp runner script like we used to.

The python virtual environment can only be enabled if python >= 3.8 is found. For most of the downstream tests this wasn't the case. The gnu@fedora-37 runners did however have a more up-to-date python installation. If possible, it would be great to configure these CI tests with the option `-DENABLE_FCKIT_VENV=ON` to have CI test coverage of the virtual environment.

File: cmake/fckit_install_venv.cmake [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation nor
+# does it submit to any jurisdiction.
+
+macro( fckit_install_venv )
+
+    # Create a virtualenv
+    set( VENV_PATH ${CMAKE_CURRENT_BINARY_DIR}/fckit_venv )
+    message( STATUS "Create Python virtual environment ${VENV_PATH}" )
```
Comment (wdeconinck) [inline cmake/fckit_install_venv.cmake:13]: Please use `ecbuild_info()` here and below for logging.

File: cmake/fckit_install_venv.cmake [hunk 3]
```diff
@@ -0,0 +1,68 @@
+# (C) Copyright 2013 ECMWF.
```
Comment (wdeconinck) [inline cmake/fckit_install_venv.cmake:1]:
```suggestion
# (C) Copyright 2024 ECMWF.
```

File: src/tests/test_yaml_reader.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+
+# (C) Copyright 2013 ECMWF.
```
Comment (wdeconinck) [inline src/tests/test_yaml_reader.py:3]:
```suggestion
# (C) Copyright 2024 ECMWF.
```

**Review by wdeconinck — COMMENTED** [review]
Very nice improvements! Just a few minor edits requested.

**Review by wdeconinck — APPROVED** [review]

**Pull request merged**